releasing/plugin-commands-deploy/src/createDeployFiles.ts

```diff
@@ -0,0 +1,262 @@
+        patchedDependencies: undefined,
+        packageExtensions: undefined, // the effects of the package extensions should already be part of the package snapshots
+      },
+    },
+  }
+
+  if (lockfile.patchedDependencies) {
+    result.lockfile.patchedDependencies = {}
+    result.manifest.pnpm!.patchedDependencies = {}
+
+    for (const name in lockfile.patchedDependencies) {
+      const patchInfo = lockfile.patchedDependencies[name]
+      const resolvedPath = path.resolve(rootProjectManifestDir, patchInfo.path)
+      const relativePath = normalizePath(path.relative(deployDir, resolvedPath))
+      result.manifest.pnpm!.patchedDependencies[name] = relativePath
+      result.lockfile.patchedDependencies[name] = {
+        hash: patchInfo.hash,
+        path: relativePath,
+      }
+    }
+  }
+
+  return result
+}
+
+interface ConvertOptions {
+  allProjects: CreateDeployFilesOptions['allProjects']
+  deployedProjectRealPath: string
+  projectRootDirRealPath: string
+  lockfileDir: string
+}
+
+function convertPackageSnapshot (inputSnapshot: PackageSnapshot, opts: ConvertOptions): PackageSnapshot {
+  const dependencies = convertResolvedDependencies(inputSnapshot.dependencies, opts)
+  const optionalDependencies = convertResolvedDependencies(inputSnapshot.optionalDependencies, opts)
+  return {
+    ...inputSnapshot,
+    dependencies,
+    optionalDependencies,
+  }
+}
+
+function convertProjectSnapshotToPackageSnapshot (
+  projectSnapshot: ProjectSnapshot,
+  opts: ConvertOptions & { deployDir: string }
+): PackageSnapshot {
+  const resolution: DirectoryResolution = {
+    type: 'directory',
+    directory: normalizePath(path.relative(opts.deployDir, opts.projectRootDirRealPath)),
+  }
+  const dependencies = convertResolvedDependencies(projectSnapshot.dependencies, opts)
+  const optionalDependencies = convertResolvedDependencies(projectSnapshot.optionalDependencies, opts)
+  return {
+    dependencies,
+    optionalDependencies,
+    resolution,
+  }
+}
+
+function convertResolvedDependencies (input: ResolvedDependencies | undefined, opts: ConvertOptions): ResolvedDependencies | undefined {
+  if (!input) return undefined
+  const output: ResolvedDependencies = {}
+
+  for (const key in input) {
+    const spec = input[key]
+    const depRealPath = resolveLinkOrFile(spec, opts)
+    if (!depRealPath) {
+      output[key] = spec
+      continue
+    }
+
+    if (depRealPath === opts.deployedProjectRealPath) {
+      output[key] = 'link:.' // the path is relative to the lockfile dir, which means '.' would reference the deploy dir
+      continue
+    }
+
+    output[key] = createFileUrlDepPath(depRealPath, opts.allProjects)
+  }
+
+  return output
+}
+
+function resolveLinkOrFile (spec: string, opts: Pick<ConvertOptions, 'lockfileDir' | 'projectRootDirRealPath'>): string | undefined {
+  // try parsing `spec` as `spec(peers)`
+  const hasPeers = /^(?<spec>[^()]+)(?<peers>\(.+\))$/.exec(spec)
```

Comment:
you don't need to write new parsers. There are already parsers for depPath in `@pnpm/dependency-path`

Reply:
I can't make it work. I have published the change to a branch named [try-using-dependency-path](https://github.com/pnpm/pnpm/tree/try-using-dependency-path).

latest commit: https://github.com/pnpm/pnpm/commit/0dce3be50463879ce81abbb76571f7b476cbd4bb

diff: https://github.com/pnpm/pnpm/compare/deploy-prod-fails-due-to-missing-dev-8778...try-using-dependency-path

Reply:
The test regarding peer suffix fails. Attempting to deploy `@pnpm/headless` from `pnpm@v8` also fails.

Reply:
`parseDepPath` doesn't work either: https://github.com/pnpm/pnpm/tree/try-using-dependency-path-2

Reply:
> The test regarding peer suffix fails. Attempting to deploy @pnpm/headless from pnpm@v8 also fails.

I will need a few hours to look into it, which is problematic now. I will try to look into it in the next couple of days.